pkg/apis/core/v1/helper/helpers_test.go

```diff
@@ -607,109 +605,97 @@ func TestMatchNodeSelectorTerms(t *testing.T) {
 		{
 			name: "nil terms",
 			args: args{
-				nodeSelectorTerms: nil,
-				nodeLabels:        nil,
-				nodeFields:        nil,
+				nodeSelector: nil,
+				node:         nil,
 			},
 			want: false,
 		},
 		{
 			name: "node label matches matchExpressions terms",
 			args: args{
-				nodeSelectorTerms: []v1.NodeSelectorTerm{
+				nodeSelector: &v1.NodeSelector{NodeSelectorTerms: []v1.NodeSelectorTerm{
 					{
 						MatchExpressions: []v1.NodeSelectorRequirement{{
 							Key:      "label_1",
 							Operator: v1.NodeSelectorOpIn,
 							Values:   []string{"label_1_val"},
 						}},
 					},
-				},
-				nodeLabels: map[string]string{"label_1": "label_1_val"},
-				nodeFields: nil,
+				}},
+				node: &v1.Node{ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"label_1": "label_1_val"}}},
 			},
 			want: true,
 		},
 		{
 			name: "node field matches matchFields terms",
 			args: args{
-				nodeSelectorTerms: []v1.NodeSelectorTerm{
+				nodeSelector: &v1.NodeSelector{NodeSelectorTerms: []v1.NodeSelectorTerm{
 					{
 						MatchFields: []v1.NodeSelectorRequirement{{
 							Key:      "metadata.name",
 							Operator: v1.NodeSelectorOpIn,
 							Values:   []string{"host_1"},
 						}},
 					},
-				},
-				nodeLabels: nil,
-				nodeFields: map[string]string{
-					"metadata.name": "host_1",
-				},
+				}},
+				node: &v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "host_1"}},
 			},
 			want: true,
 		},
 		{
 			name: "invalid node field requirement",
 			args: args{
-				nodeSelectorTerms: []v1.NodeSelectorTerm{
+				nodeSelector: &v1.NodeSelector{NodeSelectorTerms: []v1.NodeSelectorTerm{
 					{
 						MatchFields: []v1.NodeSelectorRequirement{{
 							Key:      "metadata.name",
 							Operator: v1.NodeSelectorOpIn,
 							Values:   []string{"host_1, host_2"},
 						}},
 					},
-				},
-				nodeLabels: nil,
-				nodeFields: map[string]string{
-					"metadata.name": "host_1",
-				},
+				}},
+				node: &v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "host_1"}},
 			},
 			want: false,
 		},
 		{
 			name: "fieldSelectorTerm with node labels",
 			args: args{
-				nodeSelectorTerms: []v1.NodeSelectorTerm{
+				nodeSelector: &v1.NodeSelector{NodeSelectorTerms: []v1.NodeSelectorTerm{
 					{
 						MatchFields: []v1.NodeSelectorRequirement{{
 							Key:      "metadata.name",
 							Operator: v1.NodeSelectorOpIn,
 							Values:   []string{"host_1"},
 						}},
 					},
-				},
-				nodeLabels: map[string]string{
+				}},
+				node: &v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "not_host_1", Labels: map[string]string{
```

Comment:
Note: the addition of `Name` in this test is due to the discussion around https://github.com/kubernetes/kubernetes/pull/95871#discussion_r512076139, concerning whether we should check matchfields only if `node.Name` is set.

In practical cases, this makes sense. Current callers that are not actually passing in a node object don't care about if `MatchFields` passes, and because the checks of `MatchExpressions` and `MatchFields` are ORed it can be ignored.

In this test case, we are specifically checking that a MatchFields block won't return true for matching Labels. In a real use case for this, the caller would either be passing in a node with a name, or also checking against a list of labels.